src/acrquery/azext_acrquery/tests/latest/test_acr_query.py

```diff
@@ -0,0 +1,47 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.testsdk import ScenarioTest
+from azure.cli.command_modules.acr._docker_utils import (
+    EMPTY_GUID
+)
+
+REGISTRY_NAME = "metadataunittest"
+RESOURCE_GROUP = "cabarker"
+REPOSITORY_NAME = "nginx"
+
+
+class AcrQueryTests(ScenarioTest):
+
+    def test_acrquery(self):
+        # Query with bearer auth and filter by count
+        credentials = self.cmd(
+            'acr credential show -n {} -g {}'.format(REGISTRY_NAME, RESOURCE_GROUP)).get_output_in_json()
```

Comment:
In a test, frequently used parameters could be input firstly, as:
self.kwargs.update({
    'acrName': REGISTRY_NAME
})
and later the {acrName} could be used in command directly:
self.cmd('acr credential show -n {acrName} -g {rg}')
It is suggested to use @ResourceGroupPreparer(name_prefix='xxx', location='xxx') as well, since this decorator is used to create a resource group for the test and will delete it automatically after the test finishes running.

you could refer to the test in https://github.com/Azure/azure-cli-extensions/blob/main/src/image-copy/azext_imagecopy/tests/latest/test_image_copy_commands.py

Reply:
Thanks for the feedback @yanzhudd! I've added the self.kwargs at the top of the test method. This test doesn't need to create/delete an RG so I did not add the ResourceGroupPreparer. 

One note about the self.kwargs is that at lines such as 
 [this one](https://github.com/CarolineNB/azure-cli-extensions/blob/0d67b52a8d005536b4d38d500af3fc6de80d009e/src/acrquery/azext_acrquery/tests/latest/test_acr_query.py#L30) mix both kwargs and variables, so it doesnt recognize kwargs arguments with .format. Instead, I manually wrote out the variable.

Reply:
Hi @CarolineNB 
this case could be resloved by adding self.kwargs.update at the middle of the test or if just one parameter it could be wriiten as:
self.kwargs['userName'] = userName